cmd/goal/application.go

```diff
@@ -928,17 +929,148 @@ func printable(str string) bool {
 	return true
 }
 
+func jsonPrintable(str string) bool {
+	// htmlSafeSet holds the value true if the ASCII character with the given
+	// array position can be safely represented inside a JSON string, embedded
+	// inside of HTML <script> tags, without any additional escaping.
+	//
+	// All values are true except for the ASCII control characters (0-31), the
+	// double quote ("), the backslash character ("\"), HTML opening and closing
+	// tags ("<" and ">"), and the ampersand ("&").
+	var htmlSafeSet = [utf8.RuneSelf]bool{
+		' ':      true,
+		'!':      true,
+		'"':      false,
+		'#':      true,
+		'$':      true,
+		'%':      true,
+		'&':      false,
+		'\'':     true,
+		'(':      true,
+		')':      true,
+		'*':      true,
+		'+':      true,
+		',':      true,
+		'-':      true,
+		'.':      true,
+		'/':      true,
+		'0':      true,
+		'1':      true,
+		'2':      true,
+		'3':      true,
+		'4':      true,
+		'5':      true,
+		'6':      true,
+		'7':      true,
+		'8':      true,
+		'9':      true,
+		':':      true,
+		';':      true,
+		'<':      false,
+		'=':      true,
+		'>':      false,
+		'?':      true,
+		'@':      true,
+		'A':      true,
+		'B':      true,
+		'C':      true,
+		'D':      true,
+		'E':      true,
+		'F':      true,
+		'G':      true,
+		'H':      true,
+		'I':      true,
+		'J':      true,
+		'K':      true,
+		'L':      true,
+		'M':      true,
+		'N':      true,
+		'O':      true,
+		'P':      true,
+		'Q':      true,
+		'R':      true,
+		'S':      true,
+		'T':      true,
+		'U':      true,
+		'V':      true,
+		'W':      true,
+		'X':      true,
+		'Y':      true,
+		'Z':      true,
+		'[':      true,
+		'\\':     false,
+		']':      true,
+		'^':      true,
+		'_':      true,
+		'`':      true,
+		'a':      true,
+		'b':      true,
+		'c':      true,
+		'd':      true,
+		'e':      true,
+		'f':      true,
+		'g':      true,
+		'h':      true,
+		'i':      true,
+		'j':      true,
+		'k':      true,
+		'l':      true,
+		'm':      true,
+		'n':      true,
+		'o':      true,
+		'p':      true,
+		'q':      true,
+		'r':      true,
+		's':      true,
+		't':      true,
+		'u':      true,
+		'v':      true,
+		'w':      true,
+		'x':      true,
+		'y':      true,
+		'z':      true,
+		'{':      true,
+		'|':      true,
+		'}':      true,
+		'~':      true,
+		'\u007f': true,
+	}
+
+	for _, r := range str {
+		if r >= utf8.RuneSelf {
+			return false
+		}
+		if htmlSafeSet[r] == false {
+			return false
+		}
+	}
+	return true
+}
+
 func heuristicFormatStr(str string) string {
-	if printable(str) {
+	// if it's a base32 address string which has a valid checksum, then make the assumption it's an address
```

Comment:
I believe this is now correct as intended, but I also think it doesn't add anything. Everything in the base32 set is in the jsonPrintable set, so falling through to the next clause would be as good. If you still want to explicitly call out that addresses should pass through, I won't block it.

Reply:
The trigger for this change was the `goalAppAccountAddressTest` which was failing. It was failing because *some* of the valid addresses were a valid unicode printable strings, but invalid json printable strings.
as a result, the output of `goal ...` used to contain "encoded" entries for some of the account addresses.

I'm going to remove the top section completely, and leave only the json printing test, as it seems to suffice.